pyo3-macros-backend/src/frompyobject.rs

```diff
@@ -281,10 +281,20 @@ impl<'a> Container<'a> {
                 value: expr_path,
             }) = from_py_with
             {
+                let deprecation = expr_path.from_lit_str.then(|| {
+                    quote_spanned! { expr_path.span() =>
+                        #[deprecated(since = "0.24.0", note = "`from_py_with` string literals are deprecated. Use the function path instead.")]
+                        #[allow(dead_code)]
+                        const LIT_STR_DEPRECATION: () = ();
+                        let _: () = LIT_STR_DEPRECATION;
+                    }
```

Comment:
To avoid repetition, shall we put a helper function somewhere to produce these fragments?

Also, I wonder if we should be more direct with the hint to users. E.g. 

```
= help: use `foo` instead of `"foo"`
```
(we could even substitute in the user input)

or maybe 
```
remove the quotes from the literal
```

Reply:
Good idea! I make a small helper in `utils` and changed the deprecation message, let me know what you think